enhancements/sig-policy/74-order-policy-execution/README.md

```diff
@@ -0,0 +1,219 @@
+
+- Performance hit due to having to watch extra resources (utilizing the watch
+library should help with this)
+- Policies that are dependent on another policy status to be processed will
+take longer to show status than normal policies. This shouldn't be a big issue
+most of the time, but if the config policy controller is at its saturation point
+(500+ configuration policies) it could take over a minute for the dependent policy
+status to update correctly
+
+## Design Details
+
+Here is what the new spec field will look like in a policy. The top-level
+`dependencies` field applies to all policy templates in the policy, while
+the template-level `extraDependencies` field can be used to set up
+dependencies for a specific policy template. The top-level `dependencies`
+will take priority over the template-level `extraDependencies`, so policy
+templates that have `extraDependencies` satisfied but not `dependencies`
+will still be marked as `pending`.
+
+```
+apiVersion: policy.open-cluster-management.io/v1
+kind: Policy
+metadata:
+  name: my-complicated-policy
+spec:
+  dependencies: //for all policy templates
+    - kind: Policy
+      name: namespace-foo-setup-policy
+      compliance: Compliant
+  policy-templates:
+    - extraDependencies: //for this policy template only
+        - kind: ConfigurationPolicy
+          name: bar
+          compliance: NonCompliant
+      objectDefinition:
+        apiVersion: policy.open-cluster-management.io/v1
+        kind: ConfigurationPolicy
+        metadata:
+          name: fix-bar-in-foo
+        spec:
+          remediationAction: enforce
+          namespaceSelector:
+            include: ["foo"]
+          object-templates:
+            ...
+```
+
+When the dependencies for a policy template to be evaluated are not met, we plan to add logic
+in the template sync portion of the policy framework addon that will ignore those templates
+until they are ready. Putting this logic in the template sync ensures that whether a policy
+template is evaluated will be determined on the managed cluster level rather than the hub
+level, which is ideal because policy templates on different managed clusters may need to be
+processed differently based on the state of each managed cluster.
+
+If a policy template meets its dependencies and gets processed properly, but at some point
+afterward the state of the cluster changes and those dependencies are no longer met, we do
+not want the template to be active anymore. Utilizing the watch library should allow us to
+follow the dependencies for each template and put the template back into a pending state
+when this happens. When the template is put into a pending state, the configuration policy
+will be deleted, so if a dependency is no longer satisfied, any objects created
+on the cluster as a result of that dependency will be deleted.
+
+### Open Questions
+
+Is setting an `apiGroup` field in `dependencies/extraDependencies` necessary? Currently we should
```

Comment:
What if we add a new alpha or beta policy type? Is the plan to use [restmapper](https://pkg.go.dev/k8s.io/client-go/restmapper)?

Reply:
I think the best way to handle this would be if we want to add a new alpha or beta policy type, we will also add the `apiGroup` field here

Reply:
@mprahl what do you mean by new alpha or beta policy type?

Reply:
@JustinKuli say I create a new policy with the kind `OperatorPolicy` and it's in the `policy.open-cluster-management.io` `apiGroup` but it's a `beta1` version. How will this be handled by the dependency resolution? When reading this doc, I got the impression that `v1` was always implied.

Reply:
Ah, ok. Good point: since we want to support PolicySet, and that's only `v1beta1` right now, we probably already need an apiGroup field here.

Reply:
Maybe the field should be `apiVersion` to match what you specify in a Kubernetes manifest?

Reply:
In the call today I said `PolicySet` would work... but those aren't copied to managed clusters, are they? So they probably can't be used as a dependency.

Reply:
@mprahl shared the idea that the propagator could see a PolicySet as a dependency, and replace it with the list of Policies in that set. We might need to be careful to keep that replacement updated if the PolicySet changes, but I think overall it would work.

I was concerned it might be confusing if the replicated policy and the root policy had different specs, but we already have that happen with hub templates, so it's really not that bad. And it would be more clear what exactly the policy is waiting on (a specific policy, not a PolicySet that the user would need to look up)